kubernetes/charts/weblogic-operator/templates/_operator-dep.tpl

```diff
@@ -165,3 +167,170 @@ spec:
             path: logstash.conf
       {{- end }}
 {{- end }}
+---
+  {{ $chartVersion := .Chart.Version }}
+  {{ $webhookExists := include "utils.verifyExistingWebhookDeployment" (list $chartVersion) | trim }}
+  {{- if or (ne $webhookExists "true") (.recreateWebhook) }}
+    # webhook does not exist or chart version is newer, create a new webhook
+    apiVersion: "v1"
+    kind: "ConfigMap"
+    metadata:
+      labels:
+        weblogic.webhookName: {{ .Release.Namespace | quote }}
+      name: "weblogic-webhook-cm"
+      namespace: {{ .Release.Namespace | quote }}
+    data:
+      serviceaccount: {{ .serviceAccount | quote }}
+---
+    # webhook does not exist or chart version is newer, create a new webhook
+    apiVersion: "apps/v1"
+    kind: "Deployment"
+    metadata:
+      name: "weblogic-operator-webhook"
+      namespace: {{ .Release.Namespace | quote }}
+      labels:
+        weblogic.webhookName: {{ .Release.Namespace | quote }}
+        weblogic.webhookVersion: {{ .Chart.Version }}
+      {{- if .preserveWebhook }}
+      annotations:
+        "helm.sh/hook": pre-install
+        "helm.sh/resource-policy": keep
+        "helm.sh/hook-delete-policy": "before-hook-creation"
+      {{- end }}
+    spec:
+      strategy:
+        type: Recreate
+      selector:
+        matchLabels:
+          weblogic.webhookName: {{ .Release.Namespace | quote }}
+      replicas: 1
```

Comment:
As described in my review note, I think that this should be "2" and we should resolve startup related concurrency issues.

Reply:
I'm not clear why we need to default to "2" copies since webhook is going to be rarely used (only when converting from v8 to v9 or vice-versa). Not sure if it's worth creating 2 pods for just this reason. The customers can always change the replica if they feel it's necessary.

Reply:
Okay, that's fair. I was thinking that if we defaulted to 2 replicas then we would actually make sure that the concurrency issues were resolved ;)